### DISCUSSION TITLE
acme.sh - Where to deploy issued cert files

### DISCUSSION THREAD

**Discussion by kw-sudo** [NginxProxyManager/nginx-proxy-manager#4039] [Q&A] [answered]
I run NPM with sqlite.db in a Docker container. To avoid having to open ports, I prefer acme.sh.

I successfully issued my cert via DNS challenge and all cert files are stored in the 'download folder'.

- ca.cer
- mydomain.com.cer
- mydomain.com.conf
- mydomain.com.csr
- mydomain.com.csr.conf
- mydomain.com.key
- fullchain.cer

I'm just not sure which deploy variant I have to choose to install the certificate in NPM so that it is recognized and automatically renewed?

There are two variants:
a) [deploy to docker containers](https://github.com/acmesh-official/acme.sh/wiki/deploy-to-docker-containers)
or
b) [Deploy ssl certs to nginx](https://github.com/acmesh-official/acme.sh/wiki/Deploy-ssl-certs-to-nginx)

Is this the right way at all or do I have to approach this completely differently with acme.sh?

Regards,
Oliver

**Comment by kw-sudo** [comment]
I played a little trick ;) Can it work this way?

I solved it using the 'cert docker to docker' variant.

NPM creates the custom certs under `/data/custom_ssl/npm-x`. Now I simply created a directory 'npm-1' manually and specified this path as the target in the deploy book.

Then I updated the NPM webGUI and it accepted the cert with the corresponding validity. 

The renew process is initiated by acme.sh, so that might work?

**Comment by kw-sudo** [comment, marked as answer]
I took a wrong turn and am now using the DNS challenge from NPM. acme.sh is no longer necessary.

**Comment by kw-sudo** [comment]
cleared

**Comment by jmtremblay2** [comment]
Related, but wanted to share

I wanted to programmatically update my certificates that I get from my registrar. Currently testing but my idea is to 
1) from the data folder, read the expiration of the current certificate for my domain
2) if close to expiration, call my registrar's API to get new certs.
3) if new certificates have been issues, update the custom-ssl folder with my new certificates
4) write a new expires_on in the database.

https://github.com/jmtremblay2/porkbun/blob/certificates/retrieve_ssl.py
